compiler/rustc_builtin_macros/src/autodiff.rs

```diff
@@ -0,0 +1,707 @@
+#![allow(unused_imports)]
```

Comment:
It's not about development, it's just that in one config (llvm_enzyme) they will be needed, in the other they won't.
Rustc apparently isn't smart enough to understand that. In some modules I created two submodules and based on the config I imported things from either of the two, I could try to do that here too?

Reply:
Ah right, `unused_imports` don't necessarily account for `cfg`.

In this case, I would recommend that we use lint-expectations, i.e. `#[expect(unused_imports)]` on the pub items that are still flagged by `unused_imports`, in case one day we make `unused_imports` sufficiently smart to account for `cfg`s.